### ISSUE TITLE
STM API lost in expressiveness

### ISSUE THREAD

**Issue by gares** [coq/coq#7279]
PR #1087 made the API functional, removed the `backup` and `restore` functions.
The comment says

    For now only, the API is functional; a PR switching the internals
    should come soon thou; however we must first fix some initialization
    bugs.

pidetop uses (in a crucial way) `backup` and `restore`, and now I can't port it to 8.8.0 since the current API exposes a `~doc` you can pass around but that is completely ignored.

I propose to add back `backup` and `restore` in `stm.mli`

Bottom line: we should not accept any more PRs based on "will be fixed later" since we suck at tracking that.

**Comment by ejgallego** [comment]
This was closed by #7280 , also #7281 and #7282 are related.

**Comment by ejgallego** [comment]
Well, maybe the closing of the bug is premature as it has not been merged in 8.8.1 yet.

**Comment by gares** [comment]
It will be closed automatically when #7280 is merged if we mention it in there

**Comment by gares** [comment]
oops, race condition
